### PR TITLE
[TIMOB-19957] Implement Ti.UI.ProgressBar.backgroundColor

### PR DESCRIPTION
**Pull request by garymathews**
- Implement ```Titanium.UI.ProgressBar.backgroundColor```
- Implement ```Titanium.UI.ProgressBar.backgroundDisabledColor```

###### TEST CASE
```Javascript
var win = Ti.UI.createWindow({}),
    progressBar = Ti.UI.createProgressBar({top: 100, backgroundColor:'green', backgroundDisabledColor:'red'});

win.add(progressBar);
win.open();

```

[JIRA Ticket](https://jira.appcelerator.org/browse/TIMOB-19957)

### CODE REVIEW

File: Source/UI/src/ProgressBar.cpp [hunk 1]
```diff
@@ -109,5 +120,18 @@ namespace TitaniumWindows
 			bar__->Foreground = ref new Windows::UI::Xaml::Media::SolidColorBrush(WindowsViewLayoutDelegate::ColorForName(colorName));
 		}
 
+		void ProgressBar::set_backgroundColor(const std::string& colorName) TITANIUM_NOEXCEPT
+		{
+			Titanium::UI::ProgressBar::set_backgroundColor(colorName);
+			backgroundColorBrush__ = ref new Windows::UI::Xaml::Media::SolidColorBrush(WindowsViewLayoutDelegate::ColorForName(colorName));
+			bar__->Background = backgroundColorBrush__;
```
Comment (sgtcoolguy) [inline Source/UI/src/ProgressBar.cpp]: We likely need to check if bar->IsEnabled here before setting it?

File: Source/UI/src/ProgressBar.cpp [hunk 2]
```diff
@@ -109,5 +120,18 @@ namespace TitaniumWindows
...
+		{
+			Titanium::UI::ProgressBar::set_backgroundColor(colorName);
+			backgroundColorBrush__ = ref new Windows::UI::Xaml::Media::SolidColorBrush(WindowsViewLayoutDelegate::ColorForName(colorName));
+			bar__->Background = backgroundColorBrush__;
+		}
+		
+		void ProgressBar::set_backgroundDisabledColor(const std::string& colorName) TITANIUM_NOEXCEPT
+		{
+			Titanium::UI::ProgressBar::set_backgroundDisabledColor(colorName);
+			backgroundDisabledColorBrush__ = ref new Windows::UI::Xaml::Media::SolidColorBrush(WindowsViewLayoutDelegate::ColorForName(colorName));
```
Comment (sgtcoolguy) [inline Source/UI/src/ProgressBar.cpp]: Check if bar__->IsEnabled here and if not, set the bar__->Background to this brush?

File: Source/UI/src/ProgressBar.cpp [hunk 3]
```diff
@@ -109,5 +120,22 @@ namespace TitaniumWindows
 			bar__->Foreground = ref new Windows::UI::Xaml::Media::SolidColorBrush(WindowsViewLayoutDelegate::ColorForName(colorName));
 		}
 
+		void ProgressBar::set_backgroundColor(const std::string& colorName) TITANIUM_NOEXCEPT
+		{
+			if (bar__->IsEnabled) {
```
Comment (sgtcoolguy) [inline Source/UI/src/ProgressBar.cpp]: This if guard should only be around the actual bar__->Background setter code. I think e still want to create the brush and set the property internally.

File: Source/UI/src/ProgressBar.cpp [hunk 4]
```diff
@@ -109,5 +120,22 @@ namespace TitaniumWindows
...
+				Titanium::UI::ProgressBar::set_backgroundColor(colorName);
+				backgroundColorBrush__ = ref new Windows::UI::Xaml::Media::SolidColorBrush(WindowsViewLayoutDelegate::ColorForName(colorName));
+				bar__->Background = backgroundColorBrush__;
+			}
+		}
+		
+		void ProgressBar::set_backgroundDisabledColor(const std::string& colorName) TITANIUM_NOEXCEPT
+		{
+			if (!bar__->IsEnabled) {
```
Comment (sgtcoolguy) [inline Source/UI/src/ProgressBar.cpp]: Again, we want to retain the color and brush, just check if it's not enabled and if not, then call bar__->Background = backgroundDisabledColorBrush__;

**Comment by infosia** [comment]
Sorry for late reply but I just noticed that background color should have been implemented in [WindowsViewLayoutDelegate.cpp#L1093](https://github.com/appcelerator/titanium_mobile_windows/blob/master/Source/UI/src/WindowsViewLayoutDelegate.cpp#L1093). Because Xaml ProgressBar is subclass of Control, this logic should be working now but you mean it's not? I'm interested in why WindowsViewLayoutDelegate does not work...

**Comment by infosia** [comment]
I've confirmed that it's already implemented in [WindowsViewLayoutDelegate](https://github.com/appcelerator/titanium_mobile_windows/blob/master/Source/UI/src/WindowsViewLayoutDelegate.cpp#L1093) at some point (maybe [TIMOB-19921](https://jira.appcelerator.org/browse/TIMOB-19921)) . So I'm afraid to tell you I would close this ticket for now. I see that it does not work quite well due to the layout issue, but it should be another ticket. I'll open new ticket for it soon.

**Pull request closed without merging**